### DISCUSSION TITLE
TYWE3S With different pinout

### DISCUSSION THREAD

**Discussion by Darozas** [arendst/Tasmota#16500] [Show and tell]
Hi All 

After 3 weeks without sleep finaly managed to get this unit to work.

Any one that struggling to flash TYW3S with any flasher existing on earth. Note its not flasher its TYW3S that have completely different pinout.

Now got my Light switches Tazmotised nice !!!

Pleases see below.

![image](https://user-images.githubusercontent.com/19796221/224806644-999b079f-85d5-45fc-8ade-21364cf40e70.png)

Regards

**Comment by sfromis** [comment]
Is that a photo of your module? It would appear not.... Where did you find it?

There may be similar form factor modules which are not actually TYWE3S, just appearing to be similar?

**Reply by Darozas** [reply]
I purchased switches on ebay and they came with TYWE3S chips that i could not flash using existing pinouts after 3 weeks i took steel shell off and traced pins that was completely different to all available pinouts. And i have seen loads of post that people cant flash these chips this could be the reason that Tuya changed pinout on purpose to stop replacements.

Picture attached is pinout on my module to be able flash it successfully.

Hopefully this will help alot of people go round in circles   


![20220912_181632](https://user-images.githubusercontent.com/19796221/189717021-d04c1c1c-2bd3-4551-8f14-07b3d76965e4.jpg)
![20220912_181638](https://user-images.githubusercontent.com/19796221/189717029-7a6e622c-1b29-4247-b859-694ee2f6a1e7.jpg)
![TYWE3S_pinoutDifferrent](https://user-images.githubusercontent.com/19796221/189717137-3ad1ee4e-451d-448a-8f9a-b20e13f4bab4.png)

**Comment by Darozas** [comment]
I have managed to swap TYWE3S with ESP12F by cutting trace on pad 4(esp12F GPOI0) and remapped to different pin as pin4 is not GP0i0 on TYWE3S
![20220912_182350](https://user-images.githubusercontent.com/19796221/189717959-86208432-2976-439d-aa83-b3f2855f10e7.jpg)

**Comment by joewiatyahoodotcom** [comment]
Thanks for this! I just resurrected 2 Tuyas from the trash - I couldn't flash them and tossed them - so I'll try your pinout sometime this week.

**Reply by Darozas** [reply]
Glad it helped. Led me know how did u get on!!

Flashi it dont botther replacing chip as original chip works just fine

**Comment by Skrucz** [comment]
Thanks @Darozas for your astonish sacrifice. I'm just trying to imagine what it must have felt like when you figured out the connections. Your discovery today blown my mind and save my nerves!
I've attached your picture with described other pins. It's a shame that the buttons only return a falling edge transition but still nice wall switch for good price.

![TYWE3S_alter_pinout](https://user-images.githubusercontent.com/10894708/211118163-2f6c840f-d429-4579-bab9-e74a4932499a.jpg)

**Reply by Darozas** [reply]
Glad it helped!!

**Reply by Spp595** [reply]
> Thanks @Darozas for your astonish sacrifice. I'm just trying to imagine what it must have felt like when you figured out the connections. Your discovery today blown my mind and save my nerves! I've attached your picture with described other pins. It's a shame that the buttons only return a falling edge transition but still nice wall switch for good price.
> 
> ![TYWE3S_alter_pinout](https://user-images.githubusercontent.com/10894708/211118163-2f6c840f-d429-4579-bab9-e74a4932499a.jpg)
I'm a little confused, where is gnd?

**Reply by Darozas** [reply]
Its bottom right pin . Pinout is above on module

**Reply by sfromis** [reply]
> It's a shame that the buttons only return a falling edge transition

By default, Tasmota configuration is to have features for multi press and long press, leading to button reactions being "delayed". You have SetOption settings to change this to immediate action.

**Reply by Skrucz** [reply]
> > It's a shame that the buttons only return a falling edge transition
> 
> By default, Tasmota configuration is to have features for multi press and long press, leading to button reactions being "delayed". You have SetOption settings to change this to immediate action.

There is no software problem. MCU that handle touch buttons gives only single shot pulse signal on that PCB version. Measure time between double tap is possible but its no options to achive long press.

**Reply by Spp595** [reply]
I don't know what I'm doing wrong, but my device is not flashing. I soldered everything right rx To tx a tx To rx. when I press the install button I close ground and gpio0 but the program writes that I need to close them during loading.
![IMG20230307144524](https://user-images.githubusercontent.com/104263581/223385577-ea51d722-d6a7-43b2-b08e-cfe5d676dc27.jpg)
![IMG20230307144507](https://user-images.githubusercontent.com/104263581/223385592-e7b1cc14-7350-4ff0-9d51-63c4a303c69a.jpg)

**Reply by sfromis** [reply]
You have to ground gpio0 before/while applying power.

**Reply by Spp595** [reply]
> Вы должны заземлить gpio0 до/во время подачи питания.

then the device becomes inaccessible and not even displayed while in tty

**Reply by Spp595** [reply]
> > Вы должны заземлить gpio0 до/во время подачи питания.
> 
> then the device becomes inaccessible and not even displayed while in tty

I made a mistake, it shows up but it still doesn't change anything

**Reply by Spp595** [reply]
Solved, it turns out that the esp8266 that I used as a usb bridge was flashed by esphome. I connected the switch to the programmer and everything worked.

**Reply by jkwim** [reply]
> Thanks @Darozas for your astonish sacrifice. I'm just trying to imagine what it must have felt like when you figured out the connections. Your discovery today blown my mind and save my nerves! I've attached your picture with described other pins. It's a shame that the buttons only return a falling edge transition but still nice wall switch for good price.
> 
> ![TYWE3S_alter_pinout](https://user-images.githubusercontent.com/10894708/211118163-2f6c840f-d429-4579-bab9-e74a4932499a.jpg)
 
I was struggling with a TYWE3S also and stumbled upon this thread.

This is my attempt to map all the connections.

The RESET pin is pulled up with a resistor and has not been exposed to the module pins.

I haven't been able to observe anything yet out of GPIO 1 yet.

I am trying to see whether I can flash it without having to de-solder the module.

![image](https://github.com/user-attachments/assets/005f06fc-254e-4510-ba7e-625a007f0ae3)

**Comment by Spp595** [comment]
again I come back here) I can not find a suitable template for my 2 gang no neutral switch

**Reply by Darozas** [reply]
![image](https://user-images.githubusercontent.com/19796221/224804683-20b83f3c-904b-4bd0-9811-21c69eda355a.png)
i got as this dont remember which is which try both.

![image](https://user-images.githubusercontent.com/19796221/224805196-65123f99-ab3c-42cb-8fdb-beffca280cb2.png)

**Reply by sfromis** [reply]
You can often explore pins, especially for simple switch/relay devices.
https://tasmota.github.io/docs/Configuration-Procedure-for-New-Devices/

Different 2 gang devices will very often differ when it comes to pin assignment.

**Reply by Spp595** [reply]
This template came up {"NAME":"Wall Switch 2C","GPIO":[289,288,1,32,33,0,0,0,0,224,225,0,0,0],"FLAG":0,"BASE":28} , but there are still difficulties with it. If you turn on both switches at  the same time, you cannot turn them off by pressing them, you can only use the web interface.
https://user-images.githubusercontent.com/104263581/225003516-e6bf9bb6-9dd3-425a-b9bf-c326619dd364.mp4

**Comment by zmaxb** [comment]
@Darozas Thanks buddy you saved my nerves! I tried to flash for a day on the pins from the official datasheets

**Comment by alecsandes** [comment]
Dears, please help.
Trying to flash on Tywe3s
![WhatsApp Image 2023-12-14 at 23 40 30_890f1a5b](https://github.com/arendst/Tasmota/assets/19631942/c32fbf64-fac6-400d-aa5e-6c7dfc4f1eb0)

it cannot be flashed.
I bridge GPIO0 with gnd before powering up
I release the bridge 
I start the flasher

I tried also to bridge RST with GND while flashing, nothing

**Reply by alecsandes** [reply]
![image](https://github.com/arendst/Tasmota/assets/19631942/c9657d2d-7266-4612-8a79-15655a4b65e8)

**Reply by Darozas** [reply]
Try to solder it as per pinout above.ignore writings on board.
Keep vcc and gnd as you have

**Reply by alecsandes** [reply]
Tried, didn't work ...
guess, i'm gonna throw them away

**Reply by sfromis** [reply]
To flash ESP-12 (or compatible modules like TYWE3S), you generally need to ground gpio 15, and connect EN/CH_PD to vcc. These connections are typically on the main PCB, but not if you flash the module separately.

**Reply by alecsandes** [reply]
I think I am gonna renounce ... 
I have tried the different configurations posted here ... nothing works ... I have opened the cover and found that the controller is 8266EX In fact and it is very hard for me to follow the wiring to check whether the markings are coinciding with the real pinout or not.

![WhatsApp Image 2023-12-17 at 00 44 52_15ea0744](https://github.com/arendst/Tasmota/assets/19631942/d4c3e8cb-e322-4ec7-9bc6-ca3cb8809202)

Isn;t there an easier way to flash these shits? I can make it work in tuya local as I had it in the past, but I really want it off the tuya fw, so I can access it any time I change some wifi setting ...

**Reply by sognetic** [reply]
Hi, I have the exact same module (YDM_WIFI_SV7) and ran into the same problem (flashing [esphome in my case] doesn't work when detaching the board). I did, however, discover that flashing does work when plugging the board back into the dimmer (i.e. the boards with all the mains stuff). Everything still powered from the FTDI serial converter, no mains!

The board does connect to wifi, but unfortunately I'm now stuck because the Tuya MCU module won't initialize (I think maybe because it's actually a different MCU, a Puya PY32). I'd be curious if Tasmota has the same issues. Do you still have the board and could try this, @alecsandes ?

**Comment by protectivedad** [comment]
Just a heads up I received one of the US two gang touch switches (top-bottom, not side-by-side). It had the TYWE3S module. Just a silver plate no markings. I tried the real connections no go. The connections mentioned here and still no go. I removed the board board tried again. Eventually after removing the cover I found, it had the serial connections of the new with the gpio0 of the old. So to save everyone the hassle I thought I mention it here.
